packages/headless/src/features/commerce/recommendations/recommendations-slice.ts

```diff
@@ -1,31 +1,67 @@
 import {createReducer} from '@reduxjs/toolkit';
 import {
   fetchRecommendations,
-  updateRecommendationsSlotId,
+  registerRecommendationsSlot,
 } from './recommendations-actions';
-import {getRecommendationsInitialState} from './recommendations-state';
+import {
+  getRecommendationsInitialState,
+  getRecommendationsSliceInitialState,
+  RecommendationsSlice,
+} from './recommendations-state';
 
 export const recommendationsReducer = createReducer(
   getRecommendationsInitialState(),
 
   (builder) => {
     builder
-      .addCase(updateRecommendationsSlotId, (state, action) => {
-        state.slotId = action.payload.slotId;
+      .addCase(registerRecommendationsSlot, (state, action) => {
+        const slotId = action.payload.slotId;
+
+        if (slotId in state) {
+          return;
+        }
+
+        state[slotId] = buildRecommendationsSlice();
       })
       .addCase(fetchRecommendations.rejected, (state, action) => {
-        state.error = action.payload ? action.payload : null;
-        state.isLoading = false;
+        const recommendations = state[action.meta.arg.slotId];
+
+        if (!recommendations) {
```

Comment:
Here we should test that it does not throw when the `slotId` in the action payload corresponds to no recommendations in the state.

Same comment for `fetchRecommendations.fulfilled` and `fetchRecommendations.pending`